intelmq/bots/parsers/abusech/parser_ip.py

```diff
@@ -6,36 +6,97 @@
  - https://feodotracker.abuse.ch/blocklist/
  - https://zeustracker.abuse.ch/blocklist.php
 """
-
+import re
 
 import dateutil
 
 from intelmq.lib.bot import ParserBot
+from intelmq.lib import utils
+from intelmq.lib.exceptions import PipelineError
 
-SOURCE_FEEDS = {'https://feodotracker.abuse.ch/blocklist/?download=ipblocklist': 'Cridex',
-                'https://zeustracker.abuse.ch/blocklist.php?download=ipblocklist': 'Zeus',
-                'https://zeustracker.abuse.ch/blocklist.php?download=badips': 'Zeus'}
+FEEDS = {
+    'https://feodotracker.abuse.ch/downloads/ipblocklist.csv': {
+        'format': [
+            'time.source',
+            'source.ip',
+            'source.port',
+            'malware.name'
+        ],
+        'malware': 'Cridex'
+    },
+    'https://zeustracker.abuse.ch/blocklist.php?download=ipblocklist': {
+        'format': [
+            'source.ip'
+        ],
+        'malware': 'Zeus'
+    },
+    'https://zeustracker.abuse.ch/blocklist.php?download=badips': {
+        'format': [
+            'source.ip'
+        ],
+        'malware': 'Zeus'
+    }
+}
 
 
 class AbusechIPParserBot(ParserBot):
     lastgenerated = None
 
+    def parse(self, report: dict):
+        lines = utils.base64_decode(report.get("raw")).splitlines()
+        regex = re.compile(r'^#+.*')  # filter out comments
+        top_line = list(filter(lambda i: not regex.search(i), lines))[0]
+        feed = report['feed.url']
+
+        fields = top_line.split(',')
+        if len(fields) is not len(FEEDS[feed]['format']):
+            self.logger.warning("Feed '{}' has not the expected fields: {} != {}".format(feed,
+                                                                                         len(fields),
+                                                                                         len(FEEDS[feed]['format'])))
+            raise PipelineError("Abusech ip parser is not up to date with the format online")
+
+        for line in utils.base64_decode(report.get("raw")).splitlines():
+            line = line.strip()
+            if not any([line.startswith(prefix) for prefix in self.ignore_lines_starting]):
```

Comment:
`ignore_lines_starting` is an empty list for this bot, so I don't see a reason to check for it

Reply:
It was not intended to break any of the previous functionality, so its simply a copy of the super class parse method. Since this parser was not overriding the parse method.